examples/dataflow-hashpipeline/hashpipeline.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import logging
+import os
+import sys
+
+from google.cloud.dlp import DlpServiceClient
+from google.cloud import firestore
+from google.cloud import secretmanager
+from google.cloud import pubsub
+from base64 import b64decode
+from hashlib import blake2b
+
+import apache_beam as beam
+from apache_beam.io import ReadFromText
+from apache_beam.options.pipeline_options import SetupOptions
+from apache_beam.options.pipeline_options import PipelineOptions
+
+class MyOptions(PipelineOptions):
+  @classmethod
+  def _add_argparse_args(cls, parser):
+    parser.add_value_provider_argument('--input',
+      help='Input GCS bucket url')
+    parser.add_value_provider_argument('--secret_name',
+      help='Fully qualified name of the base64 encoded Hash key including project')
+    parser.add_value_provider_argument('--project_id',
+      help='Project where the Firstore, Secret and Topic live')
+    parser.add_value_provider_argument('--collection_name',
+      help='Name of the Firestore collection where hashed SSNs are stored')
+    parser.add_value_provider_argument('--salt',
+      help='Random string for salting hashes. Max 16 Bytes for BLAKE2b')
+    parser.add_value_provider_argument('--topic',
```

Comment:
I like this idea, but how would I be able to view these values from the pipeline build phase? Every time I tried to `get()` a `ValueProvider`, it failed the job since it's only accessible in the `process` or `setup` functions of a `DoFn`